### PR TITLE
fix: base-ui-components/react/popover beta package

### LINKED ISSUE #7: base-ui-components/react/popover beta package replace
**Is your feature request related to a problem? Please describe.**
 
We are using a beta version of the first major release of this package. It's not suitable for production code
 
**Describe the solution you'd like**
 
1. Consider using a non-beta version of an existing package suite
2. The `mui` suite that we are already using has a popup library that we should consider
 
**Describe alternatives you've considered**
 
* `mui` suite

### PR DESCRIPTION
**Pull request by rawadhossain**
**What kind of change does this PR introduce?**

**Issue Number:**

Fixes #7 

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

### CodeRabbit AI Review
- [x] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [ ] I have written tests for all new changes/features
- [ ] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

## Summary by CodeRabbit

* **Refactor**
  * Replaced multi-part popover with a single Material UI popover using anchor-based open/close handling for simpler popup behavior.

* **Chores**
  * Removed an external UI dependency from project dependencies.

* **Tests**
  * Expanded and reorganized tests: added coverage for extra-users popup open/close (click and Escape), avatar rendering in main and popup views, multi-user popup styling, and numerous edge-case data scenarios.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/screens/FundCampaignPledge/FundCampaignPledge.spec.tsx [hunk 1]
```diff
@@ -639,6 +654,196 @@ describe('Testing Campaign Pledge Screen', () => {
...
+
+    const missingDatesLink = new StaticMockLink([missingDatesMock]);
+    renderFundCampaignPledge(missingDatesLink);
+
+    await waitFor(() => {
+      expect(screen.getByText('Missing Dates User')).toBeInTheDocument();
+      // Amount should still render with default currency fallback $
+      expect(screen.getByTestId('amountCell')).toHaveTextContent('$75');
+    });
+  });
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply src/screens/FundCampaignPledge/FundCampaignPledge.spec.tsx:780]: This is a nice point that you pointed out. But the current setup is perfectly working and do not need further changes as of now. If future modifications are needed, I will update it. For now current setup is fine.

File: src/screens/FundCampaignPledge/FundCampaignPledge.spec.tsx [hunk 2]
```diff
@@ -639,6 +654,196 @@ describe('Testing Campaign Pledge Screen', () => {
...
+    };
+
+    const noExtraLink = new StaticMockLink([noExtraUsersMock]);
+    renderFundCampaignPledge(noExtraLink);
+
+    await waitFor(() => {
+      expect(screen.getByText('Solo User')).toBeInTheDocument();
+    });
+    expect(screen.queryByTestId('moreContainer-singlePledge')).toBeNull();
+  });
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply src/screens/FundCampaignPledge/FundCampaignPledge.spec.tsx:727]: Addressed in previous commits

File: src/screens/FundCampaignPledge/FundCampaignPledge.spec.tsx [hunk 3]
```diff
@@ -639,6 +654,196 @@ describe('Testing Campaign Pledge Screen', () => {
...
+
+    const nullCampaignLink = new StaticMockLink([nullCampaignMock]);
+    renderFundCampaignPledge(nullCampaignLink);
+
+    await waitFor(() => {
+      expect(screen.getByTestId('searchPledger')).toBeInTheDocument();
+    });
+    // No rows should render; fallback overlay text should appear
+    expect(screen.getByText(translations.noPledges)).toBeInTheDocument();
+  });
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply src/screens/FundCampaignPledge/FundCampaignPledge.spec.tsx:805]: Addressed in previous commits

File: src/screens/FundCampaignPledge/FundCampaignPledge.spec.tsx [hunk 4]
```diff
@@ -842,4 +1047,212 @@ describe('Testing Campaign Pledge Screen', () => {
...
+      expect(screen.getByText('Avatar User')).toBeInTheDocument();
+    });
+
+    // Check that the image is rendered with the avatarURL
+    const mainUserContainer = screen.getByTestId('mainUser-avatarPledge-0');
+    expect(mainUserContainer).toBeInTheDocument();
+    const img = mainUserContainer.querySelector('img');
+    expect(img).toHaveAttribute('src', 'https://example.com/avatar.jpg');
+    expect(img).toHaveAttribute('alt', 'Avatar User');
+  });
```
Comment (rawadhossain) [inline src/screens/FundCampaignPledge/FundCampaignPledge.spec.tsx:1125]: Addressed in previous commits

File: src/screens/FundCampaignPledge/FundCampaignPledge.spec.tsx [hunk 5]
```diff
@@ -842,4 +1045,210 @@ describe('Testing Campaign Pledge Screen', () => {
...
+
+    await waitFor(() => {
+      expect(screen.getByText('Avatar User')).toBeInTheDocument();
+    });
+
+    const avatarImg = screen.getByRole('img', { name: 'Avatar User' });
+
+    expect(avatarImg).toHaveAttribute('src', 'https://example.com/avatar.jpg');
+    expect(avatarImg).toHaveAttribute('alt', 'Avatar User');
+  });
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply src/screens/FundCampaignPledge/FundCampaignPledge.spec.tsx:1125]: addressed in latest commit

File: src/screens/FundCampaignPledge/FundCampaignPledge.spec.tsx [hunk 6]
```diff
@@ -842,4 +1059,210 @@ describe('Testing Campaign Pledge Screen', () => {
...
+
+    // Check that the extra user with avatarURL has an img element
+    const extraUserContainer = screen.getByTestId('extraUser-0');
+    expect(extraUserContainer).toBeInTheDocument();
+    const img = extraUserContainer.querySelector('img');
+    expect(img).toHaveAttribute('src', 'https://example.com/extra-avatar.jpg');
+    expect(img).toHaveAttribute('alt', 'Extra With Avatar');
+
+    await userEvent.keyboard('{Escape}');
+  });
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply src/screens/FundCampaignPledge/FundCampaignPledge.spec.tsx:1208]: not necessary fix. can be avoided
